### PR TITLE
nchat: 3.67 → 5.2.11, move to by-name, add luftmensch-luftmensch to maintainers

### PR DESCRIPTION
**Pull request by Steinhagen**
## Description of changes

Update to the latest version.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Steinhagen** [comment]
This should be merged only after confirmation that all build checks pass.

**Review by luftmensch-luftmensch — APPROVED** [review]

**Comment by Steinhagen** [comment]
This seems to fail because the tests on darwin are not working as expected. For now, I'll disable the tests to see if everything working as expected. If someone has a darwin machine to test if this PR is working it would be appreciated.

**Comment by Steinhagen** [comment]
I can confirm that the OPENSSL_ROOT_DIR was the problem for the darwin build. I've also tested this about an hour ago and the build worked as expected.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 345606` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nchat</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Doesn't work for me on darwin:
```sh
[nix-shell:~/.cache/nixpkgs-review/pr-345606]$ results/nchat/bin/nchat
Failed to load /nix/store/f80r6rnllkdkfrw9sar1vhr13ml5lv08-nchat-5.2.11/bin/..//nix/store/f80r6rnllkdkfrw9sar1vhr13ml5lv08-nchat-5.2.11/lib/libtgchat.dylib, skipping profile.
Protocol Telegram not supported
No profiles setup, exiting.
```

**Comment by Steinhagen** [comment]
The error also occurs on Linux so is not darwin specific.

**Comment by Steinhagen** [comment]
Fixed the import error and squashed everything into a single commit.

File: pkgs/by-name/nc/nchat/package.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    hash = "sha256-LcTVftLKlzHstSaJjdtqoEmQ7kiqft9dbXRXoYjbaus=";
+  };
+
+  postPatch = ''
+    substituteInPlace lib/tgchat/ext/td/CMakeLists.txt \
+      --replace "get_git_head_revision" "#get_git_head_revision"
+    substituteInPlace lib/tgchat/CMakeLists.txt \
+      --replace-fail "list(APPEND OPENSSL_ROOT_DIR" "#list(APPEND OPENSSL_ROOT_DIR"
+    substituteInPlace src/main.cpp \
+      --replace-fail 'FileUtil::DirName(FileUtil::GetSelfPath()) + "/../" ' ""
```
Comment (wegank) [inline pkgs/by-name/nc/nchat/package.nix:33]: Does putting `"-DCMAKE_INSTALL_LIBDIR=lib"` in `cmakeFlags` work instead of this patch?

Comment (Steinhagen) [reply pkgs/by-name/nc/nchat/package.nix:33]: Looking at the code, it should work just fine. In fact, that way it's cleaner that trying to modify the source code.
I'll give it a try to confirm and if it works as expected, I'll push the modifications.

**Pull request merged**